static/app/views/replays/detail/domMutations/useDomFilters.tsx

```diff
@@ -0,0 +1,77 @@
+import {useCallback, useMemo} from 'react';
+import {browserHistory} from 'react-router';
+
+import {decodeList, decodeScalar} from 'sentry/utils/queryString';
+import type {Extraction} from 'sentry/utils/replays/hooks/useExtractedCrumbHtml';
+import {useLocation} from 'sentry/utils/useLocation';
+import {filterItems} from 'sentry/views/replays/detail/utils';
+
+type Options = {
+  actions: Extraction[];
+};
+
+type Return = {
+  items: Extraction[];
+  searchTerm: string;
+  setSearchTerm: (searchTerm: string) => void;
+  setType: (type: {value: string}[]) => void;
+  type: string[];
+};
+
+const FILTERS = {
+  type: (item: Extraction, type: string[]) =>
+    type.length === 0 || type.includes(item.crumb.type),
+
+  searchTerm: (item: Extraction, searchTerm: string) =>
+    JSON.stringify(item.html).toLowerCase().includes(searchTerm),
+};
+
+function useDomFilters({actions}: Options): Return {
```

Comment:
Seems like a lot of similar code between this hook and the console filters

Reply:
yes, same structure in all three. I'll look at what can be done. 

also, i'm mocking more data for tests. Not the most exciting but once we have the examples the next tests will be easier.